File: Google Code Jam/2018/Round 1 2018/problem3.cs
using System;

class Solution
{
    static void Main()
    {
        var numCases = int.Parse(Console.ReadLine());
        for (int cases = 0; cases < numCases; cases++)
        {
            var strs = Console.ReadLine().Split(' ');
            var n = int.Parse(strs[0]);
            var p = int.Parse(strs[1]);

            int total = 0;
            for (int i = 0; i < n; i++)
            {
                var strs_ = Console.ReadLine().Split(' ');
                var w = int.Parse(strs_[0]);
                var h = int.Parse(strs_[1]);

                total += (2 * w) + (2 * h) + (2.0 * Math.Sqrt(Math.Pow(w,2), Math.Pow(h,2)));
            }

            Console.WriteLine($"Case #{cases + 1}: {Math.Min(total, p)}");
        }
    }
}

File: Google Code Jam/2018/Round 1 2018/problem3.py
import math


def SolveCase(case):
    n, p = [int(x) for x in input().split(" ")]
    total = 0
    for _ in range(n):
        w, h = [int(x) for x in input().split(" ")]
        total += (2 * w) + (2 * h) + (2 * math.sqrt(w**2 + h**2))
    print("Case #{}: {}".format(case, min(p, total)))


numCases = int(input())
for i in range(numCases):
    SolveCase(i + 1)
